knowledge-graph-identification/java/src/main/java/org/linqs/psl/examples/kgi/Run.java

```diff
@@ -0,0 +1,319 @@
+        // dataStore = new RDBMSDataStore(new PostgreSQLDriver("psl", true));
+
+        model = new PSLModel(dataStore);
+    }
+
+    /**
+     * Defines the logical predicates used in this model.
+     */
+    private void definePredicates() {
+        model.addPredicate("Sub",ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("RSub", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("Mut", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("RMut", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("Inv", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("Domain", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("Range2", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("SameEntity", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("ValCat", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("ValRel", ConstantType.UniqueIntID, ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandCat_General", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandRel_General", ConstantType.UniqueIntID, ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandCat_CBL", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandRel_CBL", ConstantType.UniqueIntID, ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandCat_CMC", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandCat_CPL", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandRel_CPL", ConstantType.UniqueIntID, ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandCat_Morph", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandCat_SEAL", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("CandRel_SEAL", ConstantType.UniqueIntID, ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("PromCat_General", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("PromRel_General", ConstantType.UniqueIntID, ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("Cat", ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+        model.addPredicate("Rel", ConstantType.UniqueIntID, ConstantType.UniqueIntID, ConstantType.UniqueIntID);
+    }
+
+    /**
+     * Defines the rules for this model.
+     */
+    private void defineRules() {
+        log.info("Defining model rules");
+
+           model.addRules("0025 : VALCAT(B, C) & SAMEENTITY(A, B) & CAT(A, C) -> CAT(B, C) ^2"+
```

Comment:
I don't like this string handling.
Move the first string to the next line.
Indent them all by two indentations from the baseline (`model.addRules`).

Also put a space before the plus.
(Remember, all binary operators should not be touching their operands (with the exception of comma).)